pkg/resource/v1/certconfig/key.go

```diff
@@ -0,0 +1,7 @@
+package certconfig
+
+// Key is an interface that defines contract between Key implementations in CRD
+// frameworks and certconfig resource.
+type Key interface {
```

Comment:
I reasoned that separate `key` package wouldn't be useful or provide any additional benefit in here so I put it to same package. If needed (not necessarily), now it also reads properly: `certconfig.Key`.